packages/wonder-blocks-data/src/hooks/use-cached-effect.js

```diff
@@ -143,10 +143,15 @@ export const useCachedEffect = <TData: ValidCacheData>(
                 );
             }
             // We use our request fulfillment here so that in-flight
-            // requests are shared.
-            const request = RequestFulfillment.Default.fulfill(requestId, {
-                handler: interceptedHandler,
-            });
+            // requests are shared. In order to ensure that we don't share
+            // in-flight requests for different scopes, we add the scope to the
+            // requestId. This is just internal, so nothing else will care.
+            const request = RequestFulfillment.Default.fulfill(
+                `${requestId}|${scope}`,
```

Comment:
This solves the immediate issue that Project321 was experiencing. Subsequent PRs need to address this for SSR tracked requests - however, that's less urgent since we don't have different scoped uses of that yet (`useFastlyQuery` is currently the only SSR usage we have and its scope value is fixed); i.e. the code is broken, but in practice, we never hit the broken paths.

I'll fix that up in separate PRs.